examples/factor_model.py

```diff
@@ -1,48 +1,51 @@
-from cvxpy import *
-from ncvx import *
+import cvxpy as cp
+from cvxpy import norm, Variable, diag, sum_squares, Minimize, trace, Problem, Parameter, pos
+import ncvx as nc
 import numpy as np
-import random
 
-np.random.seed(1)
-random.seed(1)
+# This example is described in Section 6.5 of the NCVX paper.
 
+rng = np.random.default_rng(seed=123)
 
 # Generating problem data
-n = 12; k = n//2
-print("n =", n, ", k =", k)
+n = 12
+k = n // 2
+print(f"n = {n}, k = {k}")
 SNR = 20
-F = np.random.randn(n, k)
-D_true = np.random.exponential(1, size=(n, 1))
-Sigma_true = F.dot(F.T) + np.diag(D_true)
-variance = norm(Sigma_true, 'fro').value/(np.sqrt(n*n)*SNR)
-noise = np.random.normal(0, variance, size=(n,n))
-Sigma = Sigma_true + (noise + noise.T)/2
-
+F = rng.standard_normal(size=(n, k))
+D_true = rng.exponential(1, size=(n, 1))
+Sigma_true = F @ F.T + np.diag(D_true)
+variance = norm(Sigma_true, 'fro').value / (np.sqrt(n * n) * SNR)
+noise = rng.normal(0, variance, size=(n, n))
+Sigma = Sigma_true + (noise + noise.T) / 2
 
 # NC-ADMM heuristic
-Sigma_lr = Rank(n, n, k, M=None, symmetric=True)
-D_vec = Variable(n); D = diag(D_vec)
+Sigma_lr = nc.Rank((n, n), k, M=None, symmetric=True)
+D_vec = Variable(n)
+D = diag(D_vec)
 cost = sum_squares(Sigma - Sigma_lr - D)
-constraints = [D_vec >= 0, Sigma_lr >> 0]
+constraints = [D_vec >= 0, Sigma_lr >= 0]
 prob = Problem(Minimize(cost), constraints)
 
+
 def polish_func(sltn):
     matrix = sltn[Sigma_lr.id]
     w, V = np.linalg.eigh(matrix)
     w_sorted_idxs = np.argsort(-w)
     pos_w = w[w_sorted_idxs[:k]]
-    pos_V = V[:,w_sorted_idxs[:k]]
+    pos_V = V[:, w_sorted_idxs[:k]]
     Sigma_tmp = Variable((k, k), symmetric=True)
-    Sigma_small = pos_V*Sigma_tmp*pos_V.T
+    Sigma_small = pos_V * Sigma_tmp * pos_V.T
```

Comment:
Shouldn't this be @ multiplication?

Reply:
Ah, nice, yes it should be. Let me change that.